homeassistant/components/matter/__init__.py

```diff
@@ -132,6 +133,14 @@ async def on_hass_stop(event: Event) -> None:
         listen_task.cancel()
         raise ConfigEntryNotReady("Matter client not ready") from err
 
+    # Set default fabric
+    try:
+        await matter_client.set_default_fabric_label(
+            hass.config.location_name or "Home"
+        )
+    except (NotConnected, MatterError) as err:
+        raise ConfigEntryNotReady("Failed to set default fabric label") from err
```

Comment:
We need to do the same as above and cancel the listen task before raising.

Reply:
Hm, right. Alternatively, we could move it into the `_client_listen` function? :thinking: In a way it feels a bit wrong though.

Reply:
The best way is to use the config entry helper to create the task. But that's a separate refactor.